crates/bevy_math/src/ops.rs

```diff
@@ -0,0 +1,291 @@
+    }
+
+    #[inline(always)]
+    pub(crate) fn atanh(x: f32) -> f32 {
+        f32::atanh(x)
+    }
+}
+
+#[cfg(feature = "libm")]
+mod libm_ops {
+    #[inline(always)]
+    pub(crate) fn powf(x: f32, y: f32) -> f32 {
+        libm::powf(x, y)
+    }
+
+    #[inline(always)]
+    pub(crate) fn exp(x: f32) -> f32 {
+        libm::expf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn exp2(x: f32) -> f32 {
+        libm::exp2f(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn ln(x: f32) -> f32 {
+        // This isn't documented in `libm` but this is actually the base e logarithm.
+        libm::logf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn log2(x: f32) -> f32 {
+        libm::log2f(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn log10(x: f32) -> f32 {
+        libm::log10f(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn cbrt(x: f32) -> f32 {
+        libm::cbrtf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn hypot(x: f32, y: f32) -> f32 {
+        libm::hypotf(x, y)
+    }
+
+    #[inline(always)]
+    pub(crate) fn sin(x: f32) -> f32 {
+        libm::sinf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn cos(x: f32) -> f32 {
+        libm::cosf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn tan(x: f32) -> f32 {
+        libm::tanf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn asin(x: f32) -> f32 {
+        libm::asinf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn acos(x: f32) -> f32 {
+        libm::acosf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn atan(x: f32) -> f32 {
+        libm::atanf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn atan2(x: f32, y: f32) -> f32 {
+        libm::atan2f(x, y)
+    }
+
+    #[inline(always)]
+    pub(crate) fn sin_cos(x: f32) -> (f32, f32) {
+        libm::sincosf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn exp_m1(x: f32) -> f32 {
+        libm::expm1f(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn ln_1p(x: f32) -> f32 {
+        libm::log1pf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn sinh(x: f32) -> f32 {
+        libm::sinhf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn cosh(x: f32) -> f32 {
+        libm::coshf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn tanh(x: f32) -> f32 {
+        libm::tanhf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn asinh(x: f32) -> f32 {
+        libm::asinhf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn acosh(x: f32) -> f32 {
+        libm::acoshf(x)
+    }
+
+    #[inline(always)]
+    pub(crate) fn atanh(x: f32) -> f32 {
+        libm::atanhf(x)
+    }
+}
+
+#[cfg(feature = "libm")]
+pub(crate) use libm_ops::*;
+#[cfg(not(feature = "libm"))]
+pub(crate) use std_ops::*;
+
+/// This extension trait covers shortfall in determinacy from the lack of a `libm` counterpart
+/// to `f32::powi`. Use this for the common small exponents.
+pub(crate) trait FloatPow {
+    fn squared(self) -> Self;
+    fn cubed(self) -> Self;
+}
```

Comment:
Would it make sense to make these public? It would be useful at least for me in Avian and some other libs.